### PR TITLE
Feature/add api tests to ci

### PR DESCRIPTION
**Pull request by stewartie4**
running here: https://github.com/0chain/system_test/pull/258

### CODE REVIEW

File: deploy-0chain/action.yml [hunk 1]
```diff
@@ -313,13 +313,19 @@ runs:
...
+          cat build.log
+          echo
+
           [[ $num -eq 6 ]] && break
           sleep 10
           ((count++))
         done
         if [[ $num -ne 6 ]]
         then
           echo "::error title=Number of blobbers registered is less than 6::All blobbers not registered. Exiting run..."
```
Comment (cnlangzi) [inline deploy-0chain/action.yml:327]:
```suggestion
 echo "::error title=Number of blobbers registered is less than 6:: Only ${num}/6 blobbers are registered. Exiting run..."
```

**Review by cnlangzi — APPROVED** [review]
LGTM

File: deploy-0chain/action.yml [hunk 2]
```diff
@@ -313,13 +313,19 @@ runs:
...
+
           [[ $num -eq 6 ]] && break
           sleep 10
           ((count++))
         done
         if [[ $num -ne 6 ]]
         then
-          echo "::error title=Number of blobbers registered is less than 6::All blobbers not registered. Exiting run..."
+          echo "::error title=Number of blobbers registered is less than 6:: Only ${num}/6 blobbers are registered. Exiting run..."
+          cat build.log
```
Comment (Kishan-Dhakan) [inline deploy-0chain/action.yml:328]: Much needed!

File: run-system-tests/action.yml [hunk 3]
```diff
@@ -375,15 +416,15 @@ runs:
...
+        publish_dir: "./${{ env.BRANCH_DIR }}"
         destination_dir: "./${{ env.BRANCH_DIR }}"
         user_name: "github-actions[bot]"
         user_email: "github-actions[bot]@users.noreply.github.com"
 
     - name: "Wait for report deployment to complete"
       if: ${{ always() && env.TESTS_RAN == 'true' && inputs.deploy_report_page == 'true' }}
       shell: 'script --return --quiet --command "bash {0}"'
-      run: counter=0; while [ $counter -lt 5 ] && curl -o /dev/null -Isw '%{http_code}\n' "https://0chain.github.io/${{ github.event.repository.name }}/${{ env.BRANCH_DIR }}/${GITHUB_SHA}/index.html?t=$(date +%s)" | grep -v 200 > /dev/null; do sleep 10 && echo "Waiting for report page to deploy..." && ((counter=counter+1)); done
+      run: counter=0; while [ $counter -lt 5 ] && curl -o /dev/null -Isw '%{http_code}\n' "https://0chain.github.io/${{ github.event.repository.name }}/${{ env.BRANCH_DIR }}/${GITHUB_SHA}/cli/index.html?t=$(date +%s)" | grep -v 200 > /dev/null; do sleep 10 && echo "Waiting for report page to deploy..." && ((counter=counter+1)); done
```
Comment (Kishan-Dhakan) [inline run-system-tests/action.yml:427]: Would need one to wait for API report deployment too?

File: run-system-tests/action.yml [hunk 4]
```diff
@@ -203,14 +203,59 @@ runs:
...
+            echo "::error title=API System tests faled!::API System tests failed. Ensure tests are running against the correct images/branches and rule out any possible code issues before attempting a re-run"
+        fi
+
+        exit $api_system_tests_exit_code
+
+    - name: "Archive API Test Config & Console Output"
+      if: ${{ env.RUN_SUBSET_OF_TESTS == 'false' }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: Flaky-System-Test-${{ env.TEST_TIME }}
```
Comment (Kishan-Dhakan) [inline run-system-tests/action.yml:254]: Should this be something like API-System-Test-${{  env.TEST_TIME }}?

File: run-system-tests/action.yml (hunk 3 above)
Comment (stewartie4) [reply run-system-tests/action.yml:427]: Nope, once one folder is present they all are due to the way gh pages works

File: run-system-tests/action.yml (hunk 4 above)
Comment (stewartie4) [reply run-system-tests/action.yml:254]: Good point I'll resolve now

**Review by dabasov — APPROVED** [review]
lgtm

File: run-system-tests/action.yml (hunk 3 above)
Comment (Kishan-Dhakan) [reply run-system-tests/action.yml:427]: Oh, cool

**Pull request merged**